### ISSUE TITLE
how to code for X-Y motion if this kind of motion is COREXY or H-BOT

### ISSUE THREAD

**Issue by brianbui93** [MarginallyClever/GcodeCNCDemo#7]
Now, I'm troubled with coding the X-Y motion when my project focuses on the kind of Corexy motion please help me to some instruction about it !. Thank you !

**Comment by i-make-robots** [comment]
The makelangelo project already does what you want.

**Comment by brianbui93** [comment]
thanks a lot

**Comment by i-make-robots** [comment]
http://github.com/marginallyclever/makelangelo-firmware
change #define POLARGRAPH2 to #define COREXY and you should be good to go.
refer to our forums for tech support.

Glad we could help!

On Thu, Dec 3, 2015 at 8:27 PM, brianbui93 notifications@github.com wrote:

> thanks a lot
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/MarginallyClever/GcodeCNCDemo/issues/7#issuecomment-161871993
> .

## 

Dan Royer :: Marginally Clever http://www.marginallyclevergroup.com/ ::
Raising Robot Literacy :: +1.604.916.2281

**Comment by brianbui93** [comment]
There is a problem that I still don't understand in your code. It is the
Bresenham' algorithm to drawing straight line. I have read documents on
Internet but still troubled with them: how to pulse to step motors.
Would you mind if you can give me more detailed about this case ?

P/s: I'm so sorry because I'm a beginner.

**Comment by i-make-robots** [comment]
Tell me which part you understand.

On Fri, Dec 11, 2015 at 4:57 PM, brianbui93 notifications@github.com
wrote:

> There is a problem that I still don't understand in your code. It is the
> Bresenham' algorithm to drawing straight line. I have read documents on
> Internet but still troubled with them: how to pulse to step motors.
> Would you mind if you can give me more detailed about this case ?
> 
> P/s: I'm so sorry because I'm a beginner.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/MarginallyClever/GcodeCNCDemo/issues/7#issuecomment-164091561
> .

## 

Dan Royer :: Marginally Clever http://www.marginallyclevergroup.com/ ::
Raising Robot Literacy :: +1.604.916.2281

**Comment by brianbui93** [comment]
I refer to file "GcodeCNCDemo6AxisRumba.ino". I understand almost all of
your code except function which uses bresenham's line algorithm to move
both motors. Can you describe in detail how to apply this algorithm to
control step motor?. For example. I suppose that you only control 2 motors
X and Y simultaneously in which n1, n2 are the steps needed to operate
motor X, Y, respectively, t is the time to move both motors.To simplify, if
ok, can you summarize the algorithm in flow chart ?. Thank you very much !.

**Comment by misan** [comment]
Ever heard of Google? https://www.google.es/search?q=bresenham+flowchart&tbm=isch&tbo=u&source=univ&sa=X&ved=0ahUKEwiVoYGrrrTPAhWC7RQKHS2lBkgQsAQIHQ&biw=1190&bih=682&dpr=1.25

**Comment by i-make-robots** [comment]
Makelangelo-firmware supports hbot and corexy.

**Comment by i-make-robots** [comment]
@brianbui93 do you understand bresenham's algorithm now?  I'd like to close this issue.

**Comment by brianbui93** [comment]
Yes, I understand this. Thanks Dan Royer and everybody so much :)
